src/react/components/common/assetPreview/videoAsset.tsx

```diff
@@ -0,0 +1,180 @@
+import React from "react";
+import _ from "lodash";
+import {
+    Player, BigPlayButton, ControlBar, CurrentTimeDisplay,
+    TimeDivider, PlaybackRateMenuButton, VolumeMenuButton,
+} from "video-react";
+import { IAssetProps } from "./assetPreview";
+import { IAsset, AssetType, AssetState } from "../../../../models/applicationState";
+import { AssetService } from "../../../../services/assetService";
+import { CustomVideoPlayerButton } from "../../common/videoPlayer/customVideoPlayerButton";
+
+export interface IVideoAssetProps extends IAssetProps, React.Props<VideoAsset> {
+    autoPlay?: boolean;
+    timestamp?: number;
+    onChildAssetSelected?: (asset: IAsset) => void;
+}
+
+export interface IVideoAssetState {
+    loaded: boolean;
+}
+
+export interface IVideoPlayerState {
+    readyState: number;
+    paused: boolean;
+    seeking: boolean;
+    currentTime: number;
+}
+
+export class VideoAsset extends React.Component<IVideoAssetProps> {
+    public static defaultProps: IVideoAssetProps = {
+        autoPlay: true,
+        timestamp: 0,
+        asset: null,
+        childAssets: [],
+    };
+
+    public state: IVideoAssetState = {
+        loaded: false,
+    };
+
+    private videoPlayer: React.RefObject<Player> = React.createRef<Player>();
+
+    public render() {
+        const { autoPlay, asset } = this.props;
+        let videoPath = asset.path;
+        if (!autoPlay) {
+            videoPath += "#t=5.0";
+        }
+
+        return (
+            <Player ref={this.videoPlayer}
+                fluid={false}
+                width="100%"
+                height="100%"
+                autoPlay={autoPlay}
+                poster={""}
+                src={videoPath}
+            >
+                <BigPlayButton position="center" />
+                <ControlBar autoHide={false}>
+                    <CurrentTimeDisplay order={1.1} />
+                    <TimeDivider order={1.2} />
+                    <PlaybackRateMenuButton rates={[5, 2, 1, 0.5, 0.25]} order={7.1} />
+                    <VolumeMenuButton enabled order={7.2} />
+                    <CustomVideoPlayerButton order={8.1}
+                        accelerator="ArrowLeft"
+                        tooltip="Previous Tagged Frame"
```

Comment:
Localization

Reply:
Thanks for calling me out on all of this 👍

Reply:
done